File: 2015/D01/Python/Part1.py
fin = open("../input.in","r");
fout = open("../part1.out","w");

def main():
    line = fin.read();
    fout.write(str(solve(line)));

def solve(line):
    floor = 0;
    for x in line:
        if (x == "("): floor+=1;
        if (x == ")"): floor-=1;
    return floor;

main();

File: 2015/D01/Rust/Part1.rs
use std::fs;

fn main() {
    let fin = fs::read_to_string("../input.in").expect("Input Error");
    fs::write("../part1.out",solve(fin)).expect("Output Error");
}

fn solve(fin: String) -> String {
    let mut floor: i32 = 0;
    for c in fin.chars() {
        if (c == '(') {floor += 1;}
        if (c == ')') {floor -= 1;}
    }
    return floor.to_string();
}
